pkg/service/service.go

```diff
@@ -1554,3 +1558,76 @@ func segregateBackends(backends []lb.Backend) (activeBackends map[string]lb.Back
 
 	return activeBackends, nonActiveBackends
 }
+
+// SyncServicesOnDeviceChange finds and adds missing load-balancing entries for
+// new devices.
+func (s *Service) SyncServicesOnDeviceChange(nodeAddressing types.NodeAddressing) {
```

Comment:
Damn, if we could get rid of this hack https://github.com/cilium/cilium/blob/master/pkg/k8s/service.go#L178, then this function would be much simpler :-/

Reply:
What would it require to get rid of it?

Reply:
Teaching pkg/service that NodePort service is provisioned with two frontends (0.0.0.0 and NodeIP). I think it's too complicated for this PR though.